### PR TITLE
Convert TAPI models on a background thread

### PR DESCRIPTION
**Pull request by oehme**
This way the model builders can make progress while the result of the
previous one is being converted. Having a single thread for conversion
makes it safe to use the non-thread-safe converters and data structures.

FYI @vladsoroka @nskvortsov

### CODE REVIEW

File: plugins/gradle/tooling-extension-api/src/org/jetbrains/plugins/gradle/model/ProjectImportAction.java [hunk 1]
```diff
@@ -141,6 +148,13 @@ public void accept(@NotNull GradleBuild includedBuild) {
     if (isProjectsLoadedAction) {
       wrappedController.getModel(TurnOffDefaultTasks.class);
     }
+    myConverterExecutor.shutdown();
+    try {
+      myConverterExecutor.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
```
Comment (oehme) [inline plugins/gradle/tooling-extension-api/src/org/jetbrains/plugins/gradle/model/ProjectImportAction.java:153]: Waiting forever is the simplest strategy, but probably not the best :) What kind of handling would you prefer here?

**Comment by nskvortsov** [comment]
can we make the switch to basckground thread optional, so we can test it during the EAPs?

**Comment by oehme** [comment]
Yes, for sure. Where would I put that option? In the "Registry" or are you thinking of something else?

**Comment by nskvortsov** [comment]
I think a system property will be enough (so it can be configured using gradle.properties file).

**Comment by oehme** [comment]
Did you want it to be on or off by default?

**Comment by nskvortsov** [comment]
I think we can enable it by default in master. We can turn it off in the release branch afterward

**Comment by nskvortsov** [comment]
I have enabled this by default on our master branch. Commits are merged and are in progress on our internal CI

**Comment by oehme** [comment]
Thank you!

**Comment by oehme** [comment]
@nskvortsov the commit with the opt-in switch doesn't seem to be in 2022.1 EAP. I only see the commits on master, where the feature is enabled by default. Any chance to get the opt-in into the EAP so people can start trying it out?

**Comment by nskvortsov** [comment]
It will be available with the next EAP build, stay tuned )

**Pull request closed without merging**